jupyter-notebooks/Supplemental_PAO1_FastICA_example.ipynb

```diff
@@ -0,0 +1,1467 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Example: applying the PathCORE software to a FastICA model of the normalized _P. aeruginosa_ gene compendium."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This example is run after the normalized _P. aeruginosa_ compendium has been downloaded. You can download the compendium from the Zenodo record [here](https://zenodo.org/record/583694) or use the [shell script in the `../data` directory](https://github.com/greenelab/PathCORE-analysis/blob/master/data/download_data.sh) (keep in mind this script will download more than just the compendium). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Automatically created module for IPython interactive environment\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(__doc__)\n",
+    "\n",
+    "from functools import partial\n",
+    "import os\n",
+    "\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "from pathcore import CoNetwork\n",
+    "from pathcore import aggregate_permuted_network, \\\n",
+    "        network_edges_permutation_test\n",
+    "from pathcore import pathway_enrichment_with_overlap_correction\n",
+    "import seaborn as sns\n",
+    "from sklearn.decomposition import FastICA"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Specify data file paths"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "data_dir = os.path.join(\"..\", \"data\")\n",
+    "pao1_data_dir = os.path.join(data_dir, \"pao1_data\")\n",
+    "gene_compendium = os.path.join(pao1_data_dir, \"all-pseudomonas-gene-normalized.pcl\")\n",
+    "pathways = os.path.join(pao1_data_dir, \"pseudomonas_KEGG_terms.txt\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Other constants"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "n_features = 300\n",
+    "\n",
+    "gene_signature_std = 2.5\n",
+    "\n",
+    "n_permutations = 1000\n",
+    "permutation_test_alpha = 0.05"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "compendium_df = pd.read_table(gene_compendium)\n",
+    "compendium_df.set_index(\"Gene_symbol\", inplace=True)\n",
+    "\n",
+    "# Compute ICA\n",
```

Comment:
once this is in a separate cell, I recommend inserting [`%time`](http://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-time) magic